flink-ml-lib/src/main/java/org/apache/flink/ml/feature/minmaxscaler/MinMaxScalerModel.java

```diff
@@ -0,0 +1,179 @@
+                            return input.map(
+                                    new PredictLabelFunction(
+                                            broadcastModelKey,
+                                            getMax(),
+                                            getMIN(),
+                                            getFeaturesCol()),
+                                    outputTypeInfo);
+                        });
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+        ReadWriteUtils.saveModelData(
+                MinMaxScalerModelData.getModelDataStream(modelDataTable),
+                path,
+                new MinMaxScalerModelData.ModelDataEncoder());
+    }
+
+    /**
+     * Loads model data from path.
+     *
+     * @param env Stream execution environment.
+     * @param path Model path.
+     * @return MinMaxScalerModel model.
+     */
+    public static MinMaxScalerModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+        MinMaxScalerModel model = ReadWriteUtils.loadStageParam(path);
+        DataStream<MinMaxScalerModelData> modelData =
+                ReadWriteUtils.loadModelData(
+                        env, path, new MinMaxScalerModelData.ModelDataDecoder());
+        return model.setModelData(tEnv.fromDataStream(modelData));
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictLabelFunction extends RichMapFunction<Row, Row> {
+        private final String featureCol;
+        private MinMaxScalerModelData minMaxScalerModelData;
+        private final double max;
+        private final double min;
+        private final String broadcastKey;
+        private DenseVector maxVector;
+        private DenseVector minVector;
+
+        public PredictLabelFunction(
+                String broadcastKey, double max, double min, String featureCol) {
+            this.max = max;
+            this.min = min;
+            this.broadcastKey = broadcastKey;
+            this.featureCol = featureCol;
+        }
+
+        @Override
+        public Row map(Row row) {
+            if (minMaxScalerModelData == null) {
+                minMaxScalerModelData =
+                        (MinMaxScalerModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastKey).get(0);
+                maxVector = minMaxScalerModelData.maxVector;
+                minVector = minMaxScalerModelData.minVector;
+            }
+            DenseVector feature = (DenseVector) row.getField(featureCol);
+            DenseVector outputVector = new DenseVector(maxVector.size());
+            if (feature != null) {
```

Comment:
Should we throw an exception here if the input data does not contain the `featureCol`?

Reply:
OK

Reply:
@zhipeng93 @weibozhao Currently in all other model classes, we don't explicitly check whether the feature value is null. Should we follow the same practice here for consistency?

We could alternatively choose to consistently check whether feature value is null, and throw exception if it is null. But even if we don't check this, we will already have NullPointerException with proper stacktrace, which should be sufficient for us to locate the problem (i.e. feature value is null). I personally think it is simpler not to check this. What do you think?

Reply:
I have discussed with @zhipeng93. We think, it's better to remove this exception. I will refine the code later.